Subject: Kobject: remove kobject_register()

The function is no longer used by anyone in the kernel, and it prevents
the proper sending of the kobject uevent after the needed files are set
up by the caller.  kobject_init_and_add() can be used in its place.

Cc: Kay Sievers <kay.sievers@vrfy.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 include/linux/kobject.h |    1 -
 lib/kobject.c           |   18 ------------------
 2 files changed, 19 deletions(-)

--- a/include/linux/kobject.h
+++ b/include/linux/kobject.h
@@ -98,7 +98,6 @@ extern struct kobject * __must_check kob
 extern int __must_check kobject_rename(struct kobject *, const char *new_name);
 extern int __must_check kobject_move(struct kobject *, struct kobject *);
 
-extern int __must_check kobject_register(struct kobject *);
 extern void kobject_unregister(struct kobject *);
 
 extern struct kobject * kobject_get(struct kobject *);
--- a/lib/kobject.c
+++ b/lib/kobject.c
@@ -284,23 +284,6 @@ int kobject_add(struct kobject * kobj)
 }
 
 /**
- *	kobject_register - initialize and add an object.
- *	@kobj:	object in question.
- */
-
-int kobject_register(struct kobject * kobj)
-{
-	int error = -EINVAL;
-	if (kobj) {
-		kobject_init(kobj);
-		error = kobject_add(kobj);
-		if (!error)
-			kobject_uevent(kobj, KOBJ_ADD);
-	}
-	return error;
-}
-
-/**
  * kobject_set_name_vargs - Set the name of an kobject
  * @kobj: struct kobject to set the name of
  * @fmt: format string used to build the name
@@ -952,7 +935,6 @@ struct kset *kset_create_and_add(const c
 EXPORT_SYMBOL_GPL(kset_create_and_add);
 
 EXPORT_SYMBOL(kobject_init);
-EXPORT_SYMBOL(kobject_register);
 EXPORT_SYMBOL(kobject_unregister);
 EXPORT_SYMBOL(kobject_get);
 EXPORT_SYMBOL(kobject_put);
